### PR TITLE
wip: openshift/microshift: single workflow for all openshift/conformance e2e tests

### PR DESCRIPTION
**Pull request by ingvagabund**
Run openshift/conformance tests in a single job

**TODO**
- [ ] - e2e junit and e2e.log are uploaded to the GCS bucket (e.g. https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-kube-scheduler-operator/433/pull-ci-openshift-cluster-kube-scheduler-operator-master-e2e-aws/1554100065386106880/artifacts/e2e-aws/openshift-e2e-test/artifacts/)
- [ ] run `must-gather` step
- [ ] TBD

/hold

### CODE REVIEW

File: ci-operator/step-registry/openshift/microshift/e2e/openshift-conformance/openshift-microshift-e2e-openshift-conformance-commands.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+gcloud --quiet config set project "${GOOGLE_PROJECT_ID}"
+gcloud --quiet config set compute/zone "${GOOGLE_COMPUTE_ZONE}"
+gcloud --quiet config set compute/region "${GOOGLE_COMPUTE_REGION}"
+
+# Run openshift/conformance test suite
+# Later on a list of skipped tests can be provided and filtered out
+# from the suite using combination of "openshift-tests run openshift/conformance --dry-run | grep | openshift-tests run -f"
+cat  > "${HOME}"/run-test.sh <<'EOF'
+export KUBECONFIG=/var/lib/microshift/resources/kubeadmin/kubeconfig
+openshift-tests run openshift/conformance \
```
Comment (pacevedom) [inline ci-operator/step-registry/openshift/microshift/e2e/openshift-conformance/openshift-microshift-e2e-openshift-conformance-commands.sh:41]: Can we change this a bit? We will not be running all tests and we need some way of skipping them at will. Something along the lines of single node: https://github.com/openshift/release/blob/master/ci-operator/step-registry/single-node/e2e/test/single-node-e2e-test-commands.sh#L264-L281

Comment (ingvagabund) [reply ci-operator/step-registry/openshift/microshift/e2e/openshift-conformance/openshift-microshift-e2e-openshift-conformance-commands.sh:41]: Absolutely. I am testing the wiring now. Would be great to introduce TEST_SKIPS, TEST_SUITE, TEST_ARGS and other envs as well for better customization.

File: ci-operator/jobs/openshift/microshift/openshift-microshift-main-presubmits.yaml [hunk 2]
```diff
@@ -1,5 +1,77 @@
 presubmits:
   openshift/microshift:
+  - agent: kubernetes
+    always_run: false
```
Comment (dhellmann) [inline ci-operator/jobs/openshift/microshift/openshift-microshift-main-presubmits.yaml:4]: This means someone has to trigger it manually, right? I think that's what we want, until we have a better pass rate.

**Comment by ingvagabund** [comment]
```
{"component":"entrypoint","error":"process timed out","file":"k8s.io/test-infra/prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2022-08-02T14:07:32Z"}
```
Increasing the timeout from 2hrs10mins to 4hrs

File: ci-operator/jobs/openshift/microshift/openshift-microshift-main-presubmits.yaml (hunk 2 above)
Comment (ingvagabund) [reply ci-operator/jobs/openshift/microshift/openshift-microshift-main-presubmits.yaml:4]: I am aiming for the periodic jobs. I accidentally updated the wrong job.

**Comment by ingvagabund** [comment]
From ci-operator.log:
```
panic: the server could not find the requested resource (get etcds.operator.openshift.io cluster)

goroutine 1 [running]:
github.com/openshift/origin/pkg/synthetictests/allowedalerts.AllAlertTests({0x8d18ee0, 0xc000138008}, 0x6ffcf00)
	github.com/openshift/origin/pkg/synthetictests/allowedalerts/all.go:25 +0x2378
github.com/openshift/origin/pkg/synthetictests.testAlerts({0xc00ac92000, 0x6da5, 0x7974}, 0xc005709f00)
	github.com/openshift/origin/pkg/synthetictests/alerts.go:17 +0x5f
github.com/openshift/origin/pkg/synthetictests.StableSystemEventInvariants({0xc00ac92000, 0x6da5, 0x7974}, 0x0, 0xc00ac92000, {0x7fd61de, 0x15})
	github.com/openshift/origin/pkg/synthetictests/event_junits.go:34 +0xf3d
github.com/openshift/origin/pkg/test/ginkgo.JUnitForEventsFunc.JUnitsForEvents(0x6e24de0, {0xc00ac92000, 0xc018c40000, 0xc0066c8168}, 0xc016a95270, 0x6ffcf00, {0x7fd61de, 0x7004480})
	github.com/openshift/origin/pkg/test/ginkgo/synthentic_tests.go:30 +0x3c
github.com/openshift/origin/pkg/test/ginkgo.JUnitsForAllEvents.JUnitsForEvents({0xc016fd3648, 0x2, 0xc00ac92000}, {0xc00ac92000, 0x6da5, 0x7974}, 0x10, 0x1, {0x7fd61de, 0x15})
	github.com/openshift/origin/pkg/test/ginkgo/synthentic_tests.go:43 +0xf1
github.com/openshift/origin/pkg/test/ginkgo.(*Options).Run(0xc0012ff760, 0xc1f3880, {0x7faac4a, 0xf})
	github.com/openshift/origin/pkg/test/ginkgo/cmd_runsuite.go:423 +0x2125
main.newRunCommand.func1.1()
	github.com/openshift/origin/cmd/openshift-tests/openshift-tests.go:309 +0x2dc
main.mirrorToFile(0xc0012ff760, 0xc001a3fc20)
	github.com/openshift/origin/cmd/openshift-tests/openshift-tests.go:456 +0x17b
main.newRunCommand.func1(0xc001e3a500, {0xc0000c8e00, 0x8, 0x8})
	github.com/openshift/origin/cmd/openshift-tests/openshift-tests.go:290 +0x65
github.com/spf13/cobra.(*Command).execute(0xc001e3a500, {0xc0000c8d00, 0x8, 0x8})
	github.com/spf13/cobra@v1.2.1/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc001e3a280)
	github.com/spf13/cobra@v1.2.1/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.2.1/command.go:902
main.main.func1(0xc000bb3d00)
	github.com/openshift/origin/cmd/openshift-tests/openshift-tests.go:84 +0x8a
main.main()
	github.com/openshift/origin/cmd/openshift-tests/openshift-tests.go:85 +0x3b6
```

The code panics at https://github.com/openshift/origin/blob/release-4.10/pkg/synthetictests/allowedalerts/all.go#L25:
```
etcdAllowance, err = NewAllowedWhenEtcdRevisionChange(ctx, operatorClient)
if err != nil {
	panic(err)
}
```

**Comment by ingvagabund** [comment]
Even after updating the tags to 4.11 the openshift-tests binary still originates from the latest `release-4.10` HEAD.

**Comment by ingvagabund** [comment]
In case of the periodic job the `openshift-tests` binary is picked from the `microshift/microshift:microshift-test-binaries` image where the `github.com/openshift/origin` is pined to the `release-4.10` branch. In order to pick the `openshift-tests` binary from the `ocp/4.12:tests` image, new `libvirt-installer-openshift-test-binary` image is built and used for running the `openshift-microshift-e2e-openshift-conformance` step.

File: ci-operator/config/openshift/microshift/openshift-microshift-main.yaml [hunk 3]
```diff
@@ -1,21 +1,21 @@
 base_images:
   cli:
-    name: "4.10"
+    name: "4.11"
```
Comment (dhellmann) [inline ci-operator/config/openshift/microshift/openshift-microshift-main.yaml:3]: We're using 4.10 code for our first delivery until the kube rebase is stable in OCP and we can update to 4.11/4.12. Will this change to 4.11 have any impact on the desire to stay on 4.10 for a few more weeks?

**Pull request closed without merging**